### PR TITLE
Youtube integration error

### PR DESCRIPTION
**Pull request by xahgmah**
To avoid RuntimeError: Maximum call stack size exceeded(…)

### CODE REVIEW

File: dist/videojs-offset.js [hunk 1]
```diff
@@ -57,18 +57,22 @@ var vjsoffset = function(options) {
...
+    var update_time = new Date();
+    if (this.lustUpdateTime==undefined || update_time-this.lustUpdateTime>1000) {
+      var curr = this.currentTime();
+      this.lustUpdateTime = new Date();
+      if (curr < 0) {
         this.currentTime(0);
+        this.play();
+      }
+      if (this._offsetEnd > 0 && (curr > (this._offsetEnd - this._offsetStart))) {
+        this.pause();
```
Comment (z4y4ts) [inline dist/videojs-offset.js]: Leave only this line with `this.pause();` and remove the rest of this if-branch.
As discussed, we don't need it.

**Review by z4y4ts — APPROVED** [review]

File: dist/videojs-offset.js [hunk 2]
```diff
@@ -57,18 +57,22 @@ var vjsoffset = function(options) {
...
-      this.currentTime(0);
-      this.play();
-    }
-    if(this._offsetEnd > 0 && (curr > (this._offsetEnd-this._offsetStart))) {
-      this.pause();
-      if (!this._restartBeginning) {
-        this.currentTime(this._offsetEnd-this._offsetStart);
-      } else {
-        this.trigger('loadstart');
+    var update_time = new Date();
```
Comment (polesye) [inline dist/videojs-offset.js]: use camelCase

File: dist/videojs-offset.js [hunk 3]
```diff
@@ -57,18 +57,22 @@ var vjsoffset = function(options) {
...
+    if (this.lustUpdateTime==undefined || update_time-this.lustUpdateTime>1000) {
```
Comment (polesye) [inline dist/videojs-offset.js]: typo `lust`

**Review by polesye — CHANGES_REQUESTED** [review]

**Comment by z4y4ts** [comment]
Good commit message here would be:
> Handle 'timeupdate' event only once a second
> 
> To prevent infinite loop caused by calling setCurrentTime() inside a handler.

File: dist/videojs-offset.js [hunk 4]
```diff
@@ -57,18 +57,22 @@ var vjsoffset = function(options) {
...
-      this.currentTime(0);
-      this.play();
-    }
-    if(this._offsetEnd > 0 && (curr > (this._offsetEnd-this._offsetStart))) {
-      this.pause();
-      if (!this._restartBeginning) {
-        this.currentTime(this._offsetEnd-this._offsetStart);
-      } else {
-        this.trigger('loadstart');
+    var updateTime = new Date();
```
Comment (polesye) [inline dist/videojs-offset.js]:
```js
    if (new Date() - (this.lastTimeUpdate || 0) < 1000) return;
    this.lastTimeUpdate = new Date();
    var curr = this.currentTime();
    if (curr < 0) {
      this.currentTime(0);
      this.play();
    }
    if (this._offsetEnd > 0 && (curr > (this._offsetEnd - this._offsetStart))) {
      this.pause();
      if (!this._restartBeginning) {
        this.currentTime(this._offsetEnd - this._offsetStart);
      } else {
        this.trigger('loadstart');
        this.currentTime(0);
      }
    }
```

it should produce less diff

Comment (polesye) [reply dist/videojs-offset.js]: @xahgmah FYI^^

**Review by polesye — APPROVED** [review]
Please squash all commits and put a good commit message before merge.

**Pull request merged**